### DISCUSSION TITLE
Adding new column dynamically?

### DISCUSSION THREAD

**Discussion by vozvik** [mar10/wunderbaum#15] [General]
Hi Martin,

I've been trying to programatically add new columns. (the greater idea is allowing dragging elements from one tree grid to another one and adding new columns to the target grid if needed)

What is the best way of achieving this?

I tried adding new a element to tree.columns[] and calling
tree.setModified("any");

This does not work as the respective colElem is not found when rendering nodes.
(Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'colElem.style') in _render_status())

Demo with a button to add new column: https://codepen.io/vozvik/pen/bGKaQGV

Thanks a lot!

Regards
Viktor

**Comment by mar10** [comment]
Hi,
for you could try something like
```js
tree.updateColumns({ calculateCols: false });
tree.setModified("any", { removeMarkup: true });
```
but I think it would be better if a simple `tree.updateColumns()` worked as well, so I keep this issue open.

**Comment by vozvik** [comment]
I can confirm it works, thanks Martin!

**Reply by mar10** [reply]
I refactored a bit. After changing `tree.columns`, it should now be enough to call
```js
tree.setModified("colStructure")
// or
// tree.setModified("any")
```

**Comment by mar10** [comment]
btw, please let us use issues for this kind of topics, unless it is about general feature or architecture discussions.

**Reply by vozvik** [reply]
Sure! I was just not sure if I was doing something wrong so I thought I'd ask rather than complain.

**Comment by mar10** [comment]
Closing as resolved
